config/v1/types_infrastructure.go

```diff
@@ -706,6 +701,55 @@ type GCPResourceTag struct {
 	Value string `json:"value"`
 }
 
+// CloudLoadBalancerConfig contains IPs for API, API-Int and Ingress Cloud Load Balancers
+// +union
+// +kubebuilder:validation:XValidation:rule="has(self.dnsType) && self.dnsType == 'ClusterHosted' ? has(self.clusterHosted) && size(self.clusterHosted.apiIntLoadBalancerIPs) > 0 : true",message="when dnsType is ClusterHosted, API-Int Load Balancer IP addresses have to specified with clusterHostedDNS"
+type CloudLoadBalancerConfig struct {
+	// dnsType indicates the type of DNS solution in use within the cluster. Its default value of
+	// `PlatformDefault` indicates that the cluster's DNS is the default provided by the cloud platform.
+	// It can be set to `ClusterHosted` to bypass the configuration of the cloud default DNS. In this mode,
+	// the cluster needs to provide a self-hosted DNS solution for the cluster's installation to succeed.
+	// The cluster's use of the cloud's Load Balancers is unaffected by this setting.
+	// The value is immutable after it has been set at install time.
+	// Currently, there is no way for the customer to add additional DNS entries into the cluster hosted DNS.
+	// Enabling this functionality allows the user to start their own DNS solution outside the cluster after
+	// installation is complete. The customer would be responsible for configuring this custom DNS solution,
+	// and it can be run in addition to the in-cluster DNS solution.
+	// +default="PlatformDefault"
+	// +kubebuilder:validation:Enum="ClusterHosted";"PlatformDefault"
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="dnsType is immutable"
+	// +optional
+	// +unionDiscriminator
+	dnsType DNSType `json:"dnsType,omitempty"`
+
+	// clusterHosted holds the IP addresses of API, API-Int and Ingress Load
+	// Balancers on Cloud Platforms. The DNS solution hosted within the cluster
+	// use these IP addresses to provide resolution for API, API-Int and Ingress
+	// services.
+	// +optional
+	ClusterHosted *CloudLoadBalancerIPs `json:"clusterHosted,omitempty"`
+}
+
+type CloudLoadBalancerIPs struct {
+	// apiIntLoadBalancerIPs holds Load Balancer IPs for API-Int.
+	// +kubebuilder:validation:Format=ip
+	// +listType=set
+	ApiIntLoadBalancerIPs []IP `json:"apiIntLoadBalancerIPs"`
+
+	// apiLoadBalancerIPs holds API Load Balancer IPs.
+	// Could be empty for private clusters.
+	// +kubebuilder:validation:Format=ip
+	// +listType=set
+	// +optional
+	ApiLoadBalancerIPs []IP `json:"apiLoadBalancerIPs,omitempty"`
+
+	// ingressLoadBalancerIPs holds IPs for Ingress Load Balancers.
+	// +kubebuilder:validation:Format=ip
+	// +listType=set
+	// +optional
```

Comment:
What happens when this isn't supplied?

Reply:
The in-cluster DNS will not be able to serve *.apps URLs. I dodn't want to add dire sounding statements in the API.

Reply:
Doesn't need to be dire. 
```
// When not configured, may lead to issues resolving cluster ingress from within the cluster.
```